otherlibs/unix/unix.mli

```diff
@@ -394,16 +394,37 @@ val in_channel_of_descr : file_descr -> in_channel
    Text mode is supported only if the descriptor refers to a file
    or pipe, but is not supported if it refers to a socket.
 
-   On Windows: [set_binary_mode_in] always fails on channels created
-   with this function.
-
-   Beware that channels are buffered so more characters may have been
-   read from the file descriptor than those accessed using channel functions.
-   Channels also keep a copy of the current position in the file.
+   On Windows: {!Stdlib.set_binary_mode_in} always fails on channels
+   created with this function.
+
+   Beware that input channels are buffered, so more characters may
+   have been read from the descriptor than those accessed using
+   channel functions.  Channels also keep a copy of the current
+   position in the file.
+
+   Closing the channel [ic] returned by [in_channel_of_descr fd]
+   using [close_in ic] also closes the descriptor [fd].
+   Alternatively, it is correct to close the descriptor with
+   [Unix.close fd], then stop using the [ic] channel and let the GC
+   collect it.
+   Closing both the channel [ic] and the underlying descriptor [fd] is
+   incorrect and will result in a {!Stdlib.Sys_error} or {!Unix.error}
+   exception.
 
-   You need to explicitly close all channels created with this function.
-   Closing the channel also closes the underlying file descriptor (unless
-   it was already closed). *)
+   If several channels are created on the same descriptor, exactly one
+   of the channels and of the descriptor must be closed, but not
+   several.
```

Comment:
The sentence is confusing: "and of" and "several" are not so clear to me. Maybe it would be clearer to be more redundant:

> If several channels are created on the same descriptor, you should either close one of the channels (but not the others nor the descriptor) or close the descriptor (but none of the channels).

Reply:
What's wrong with "channels ⋃ {descriptor}" ?  :-)  

I simplified the explanation by omitting to mention that the descriptor can be closed.  It's not the best solution anyway.